src/symmetric_crypto/aes_256_gcm_pure/mod.rs

```diff
@@ -79,15 +102,15 @@ impl SymmetricCrypto for Aes256GcmCrypto {
 ///
 /// The total length of the encrypted data is the message length + `MAC_LENGTH`
 pub fn encrypt_combined(
-    key: &Key,
+    key: &[u8],
     bytes: &[u8],
-    nonce: &Nonce,
+    nonce: &[u8],
     additional_data: Option<&[u8]>,
 ) -> Result<Vec<u8>, CryptoCoreError> {
     let payload =
         additional_data.map_or_else(|| Payload::from(bytes), |aad| Payload { msg: bytes, aad });
-    Aes256Gcm::new(key)
-        .encrypt(GenericArray::from_slice(nonce.as_slice()), payload)
+    Aes256Gcm::new(GenericArray::from_slice(key))
+        .encrypt(GenericArray::from_slice(nonce), payload)
```

Comment:
shouldn't we remove also these `GenericArray` uses?

Reply:
Yes indeed !

Reply:
The use of `GenericArrays` in Aes was the main argument for using `GenericArrays` in CryptoCore. They are trying to switch to regular arrays [apparently](https://github.com/RustCrypto/traits/issues/970)